samples/gui/server/datauser/k8sclient.go

```diff
@@ -6,69 +6,119 @@ package datauser
 import (
 	"context"
 
-	dm "github.com/ibm/the-mesh-for-data/manager/apis/app/v1alpha1"
-
+	app "github.com/ibm/the-mesh-for-data/manager/apis/app/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/cache"
+	"k8s.io/apimachinery/pkg/types"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// DMAClient contains the contextual info about the REST client for the M4DApplication CRD
-type DMAClient struct {
-	client    *rest.RESTClient
-	namespace string
-	plural    string
-	codec     runtime.ParameterCodec
+// CreateApplication makes a new M4DApplication CRD
```

Comment:
I see a lot of changes in the GUI server, that are not clear why they are being done.

Reply:
All credentials api have been changed since they are needed to manage k8s secrets instead of communicating with vault directly. The client has been changed to allow working with more than one type of resource: m4dapplication and secret - the previous structure was bound to one resource type only.